### DISCUSSION TITLE
RFM69 continuous FSK receive

### DISCUSSION THREAD

**Discussion by obones** [jgromes/RadioLib#451] [Q&A]
Hello,

I'm working on the RFLink32 project and we have been using RadioLib to do OOK reception which works great. Basically, RadioLib sets up the RFM69 module in continuous mode and the RFLink32 code times the pulses it gets out of the DIO2 pin.
However, as I have a couple of FSK based emitters, I'm trying to replicate what is already working and so would like continuous FSK reception on DIO2.
Using AirSpy with my RTL-SDR dongle, I see these signals: 
![image](https://user-images.githubusercontent.com/1314739/150649804-9f49274b-6007-401b-a131-22e4ad316492.png)

The large red band at the bottom is a transmission by the sensor I'm after.
What I see as well is that there is something continuously beeping around 867.93MHz which seems to be seen by the RFM69 as this is the kind of signal I get with an oscilloscope on the DIO2 pin:

![image](https://user-images.githubusercontent.com/1314739/150650014-c5cd135d-e6c0-41fe-ae46-8e0fee9123c5.png)

Using Universal Radio Hacker, I have confirmed the sensor is sending a two tone FSK signal, and the rtl_433 project sees it just fine with the following description: 

> 56 us bit length with a warm-up of 1336 us mark(pulse), 1996 us space(gap), a preamble of 48 bit flips (0xaaaaaaaaaaaa) and a 0x2dd4 sync-word.

Sadly, I have a hard time finding the appropriate options and values to use with RadioLib. Here is what I have so far:

      auto radioLibModule = Module(pins::RX_CS, -1, pins::RX_RESET, -1);
      auto radio_RFM69 = new RF69(&radioLibModule);
      radio_RFM69->begin( 868.00, 9.6, 50.0F, 250.0, 12, 16);
      radio_RFM69->disableContinuousModeBitSync();
      radio_RFM69->disableSyncWordFiltering();
      radio_RFM69->disableAddressFiltering();
      radio_RFM69->setCrcFiltering(false);
      radio_RFM69->receiveDirect();

But with this, I cannot seem to get the two tones from the FSK signal decoded to the DIO2 pin.
Obviously, I have missed something here, but I can't really figure it out right now.

Do you have any ideas ? Or suggestions of things to try ?

Thanks a lot for your help

**Comment by jgromes** [comment]
Continuous mode on RF69, SX1278 and similar modules is actually pretty simple, it's just the raw output of the FSK demodulator. There's no packet handling, preamble/sync word recognition etc.

The demodulator has no idea whether it's decoding a packet, or just noise, so even without signal, there's still going to be some output on DIO2. In direct mode, it's up to the user to separate signal from noise.

See the [Direct Receive example](https://github.com/jgromes/RadioLib/blob/master/examples/SX127x/SX127x_Receive_Direct/SX127x_Receive_Direct.ino). The example is for SX1278, but the interface for RF69 is exactly the same.

**Reply by obones** [reply]
Thanks, that's what I thought, the module is working just fine, it's just that it needs to be precisely setup on the frequency and deviation.
I see on "sniffer" projects that they use a CC1101 chipset and looking at its datasheet, it seems more versatile when it comes to listening to unknown signals.
I have ordered such a module, I'll see if it picks up the signal more easily.

**Reply by jgromes** [reply]
>  it's just that it needs to be precisely setup on the frequency and deviation

That's true for any module, including CC1101.

Since you know the sync word, it should be pretty easy to use direct mode.
